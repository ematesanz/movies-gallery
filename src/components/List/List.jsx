import styles from './List.module.css';

const toggleFavorite = (index, setData) => { 
  setData(list => list.map((item, i) =>
    i === index ? { ...item, 'favorite': item.favorite ? !item.favorite : true } : item
  ));
}

const toggleWatchLater = (index, setData) => { 
  setData(list => list.map((item, i) =>
    i === index ? { ...item, 'watchLater': item.watchLater ? !item.watchLater : true } : item
  ));
}

const List = ({list, setData}) =>  {
  const listItems = list.map((item, index) => {
  const hasFavorite = item.favorite ? 'star--active' : 'star--inactive';
  const hasWatchLater = item.watchLater ? 'watch-later--active' : 'watch-later--inactive';
  const year =  item.release_date ? new Date(item.release_date).getFullYear() : '';
  return (
    <li className={styles.movie} key={item.id}>
      <div className={styles['movie__film-poster']}>
        <img className={styles.movie__img} src={`https://image.tmdb.org/t/p/w300${item['poster_path']}`} alt={item.title} />
      </div>  
      <div className={styles.movie__content}>
        <div className={styles.movie__header}>
          <h1 className={styles.movie__title}>{item.title}</h1>
          <div className={`${styles.movie__tag} ${styles['movie__tag--1']}`}>{year}</div>
          <div className={`${styles.movie__tag} ${styles['movie__tag--2']}`}>{item.vote_average}</div>
          <img className={`${styles.favorite} ${styles[hasFavorite]}`} alt='Favorite' onClick={() => toggleFavorite(index, setData)} />
          <img className={`${styles['watch-later']} ${styles[hasWatchLater]}`} alt='Watch later' onClick={() => toggleWatchLater(index, setData)} />
        </div>
        <p className={styles.movie__description}>{item.overview}</p>
      </div>
    </li>
  )});

  return <ul className={styles.movies}>{listItems}</ul>;
}

export default List;
