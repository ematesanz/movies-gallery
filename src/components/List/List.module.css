.movies{
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.movie {
  max-width: 800px;
  border-radius: 5px;
  display: flex;
  box-shadow: 0 5px 20px 10px rgba(0, 0, 0, .2);
  overflow: hidden;
}

.movie__film-poster {
  flex: 0 0 25%;
}

.movie__img {
  width: 100%;
  display: block;
}

.movie__content {
  background-color: #fff;
  flex: 1;
  padding: 35px 30px;
  display: flex;
  flex-direction: column;
}

.movie__header {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
}

.movie__price {
  background:linear-gradient(to bottom, #A9C9FF 0%, #FFBBEC 100%);
  flex: 0 0 50px;
  font-size: 18px;
  font-weight: bold;
  letter-spacing: 2px;
  color: rgb(255, 255, 255);
  writing-mode: vertical-lr;
  display: flex;
  align-items: center;
  justify-content: center;
}

.movie__title {
  font-size: 16px;
  margin-right: auto;
  color: royalblue;
}

.fa-fire {
  color: salmon;
}

.movie__tag {
  font-size: 10px;
  color: #fff;
  padding: 2px 7px;
  border-radius: 100px;
  margin-right: 8px;
  display: block;
  text-transform: uppercase;
}

.movie__tag--1 {
  background-color: #A9C9FF;
}

.movie__tag--2 {
  background-color:#FFBBEC;
}

.favorite{
  display: inline-block;
  width: 16px;
  height: 16px;
  margin-right: 5px;
  background-size: 100%;
}

img.active {
  content: url("../../img/star-active.png");
}

img.inactive {
  content: url("../../img/star-inactive.png");
}

.movie__description {
  font-size: 14px;
}