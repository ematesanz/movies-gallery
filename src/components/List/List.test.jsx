import {render} from "@testing-library/react";
import List  from "./List";

const mockData = [{
  overview: "overview",
  poster_path: "/111.jpg",
  release_date: "2022-10-19",
  title: "title"
}];
const title = "title";
const overview = "overview";

describe('List component', () => {
  it('renders correctly', () => {
    const {container, getByText} = render(
      <List list={mockData}/>
    );
    expect(container).toMatchSnapshot();
    expect(getByText(title)).toBeVisible();
    expect(getByText(overview)).toBeVisible();
  });
});