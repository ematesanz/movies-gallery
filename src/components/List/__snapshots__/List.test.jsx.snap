// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders correctly 1`] = `
<div>
  <ul
    class="movies"
  >
    <li
      class="movie"
    >
      <div
        class="movie__film-poster"
      >
        <img
          alt="title"
          class="movie__img"
          src="https://image.tmdb.org/t/p/w300/111.jpg"
        />
      </div>
      <div
        class="movie__content"
      >
        <div
          class="movie__header"
        >
          <h1
            class="movie__title"
          >
            title
          </h1>
          <div
            class="movie__tag movie__tag--1"
          >
            2022
          </div>
          <div
            class="movie__tag movie__tag--2"
          />
          <img
            alt="star"
            class="favorite inactive"
          />
        </div>
        <p
          class="movie__description"
        >
          overview
        </p>
      </div>
    </li>
  </ul>
</div>
`;
